### PR TITLE
Move from `secrecy` to `shush`

### PR DESCRIPTION
**Pull request by Eyob94**
This commit will
- Use the newly created `shush-rs` crate to replace `secrecy` adding better memory protection

### CODE REVIEW

File: examples/change_password_cli.rs [hunk 1]
```diff
@@ -21,13 +21,13 @@ async fn main() {
     use rencfs::crypto::Cipher;
     print!("Enter old password: ");
     io::stdout().flush().unwrap();
-    let old_password = SecretString::new(read_password().unwrap());
+    let old_password = SecretString::from_str(read_password().unwrap().as_str()).unwrap();
     print!("Enter new password: ");
     io::stdout().flush().unwrap();
-    let new_password = SecretString::new(read_password().unwrap());
+    let new_password = SecretString::from_str(read_password().unwrap().as_str()).unwrap();
```
Comment (radumarias) [inline examples/change_password_cli.rs]: maybe keep name as simply `from` this is how I've seen in other libs

File: examples/crypto_write_read.rs [hunk 2]
```diff
@@ -18,7 +18,7 @@ fn main() -> Result<()> {
     let cipher = Cipher::ChaCha20Poly1305;
     let mut key = vec![0; cipher.key_len()];
     crypto::create_rng().fill_bytes(key.as_mut_slice());
-    let key = SecretVec::new(key);
+    let key = SecretVec::from_vec(key);
```
Comment (radumarias) [inline examples/crypto_write_read.rs:22]: same, just `from`

File: src/crypto.rs [hunk 3]
```diff
@@ -289,7 +289,7 @@ pub fn hash_secret_string(data: &SecretString) -> [u8; 32] {
 
 #[must_use]
 pub fn hash_secret_vec(data: &SecretVec<u8>) -> [u8; 32] {
-    hash(data.expose_secret())
+    hash(data.expose_secret().as_slice())
```
Comment (radumarias) [inline src/crypto.rs]: maybe impl Deref for SecretBox and it should auto deref

**Review by radumarias — CHANGES_REQUESTED** [review]

File: examples/change_password_cli.rs (hunk 1 above)
Comment (Eyob94) [reply examples/change_password_cli.rs]: This is an implementation of the `FromStr` trait however

File: examples/crypto_write_read.rs (hunk 2 above)
Comment (Eyob94) [reply examples/crypto_write_read.rs:22]: Can do for `vec`, since this will be a breaking change on the crate itself, I'll have both, what do you think?

Comment (radumarias) [reply examples/crypto_write_read.rs:22]: then it's fine

Comment (radumarias) [reply examples/crypto_write_read.rs:22]: for 1k usages I think it's reasonable breaking change, as most of downloads are bots like from lib.rs, or increase major if you think

File: src/crypto.rs (hunk 3 above)
Comment (Eyob94) [reply src/crypto.rs]: Done

**Review by idi8-t-here — APPROVED** [review]
works like a charm, pretty awesome. Keep it up bro 💪

**Review by radumarias — APPROVED** [review]

**Pull request merged**